Update implementation guidelines for vendor (#253)
InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework: TCFv2/TCF-Implementation-Guidelines.md

Vendors must support the withdrawal of consent. Since consent is transmitted from publisher or CMPs to partners and vendors on each request, the publisher or CMP should provide a mechanism for users to withdraw consent. This mechanism may be as simple as collecting consent at each user session, or providing an option that enables the user to withdraw consent later. The UI for withdrawing consent should be the same as the UI by which consent was given.

# Vendor guidelines (DSPs, Agencies, DMPs)<a name="vendor"></a>

> For vendors or media buyers registered in the Global Vendor List, these guidelines help you understand how to determine whether you have the necessary legal bases to process a user's personal data for their purposes based on the information captured in the TC String.

Reviewer: ```suggestion
For vendors or media buyers registered in the Global Vendor List, these guidelines help you understand how to determine whether you have the necessary legal bases to process a user's personal data for the purposes you've disclosed in the GVL, based on the information contained in the TC String.
```

Author: Works for me - It might be necessary to revisit the whole guidance document.

> 1.	Vendors must be registered in the Global Vendor List if they want to read or process user data in compliance with TCF or store and/or access information on a user’s device in compliance with the TCF. 

Reviewer: ```suggestion
1.	In order to read or process user data in compliance with the TCF or store and/or access information on a user’s device in compliance with the TCF, vendors must be registered in the Global Vendor List. 
```

Author: Makes sense, this was old language anyway only change should to must initially

The status of these two fields as indicated above show that the CMP has been loaded and the user has engaged. After validating the TC data payload is suitable for your case, you should pass it in the ad call using the URL-passing macro solution detailed in documentation for the TC String.

> ## How to determine legal bases from the TC String?<a name="detlegalbasis"></a>

Reviewer: @alextcone I feel this would be a great place to integrate my flow chart, which leads to the same conclusions but being it a graphical representation is probably easier to check and thus could lead to better enforcement of this tricky part of the guidelines.
Link: https://whimsical.com/tcf-vendor-decision-flow-R2kEUj1mMLJETZRE5NH9Tu
@asr-enid I think that if you also review the flow and give it your check, you'd give Alex more confidence towards moving it forward.

Author: @Facens - sure lets do that, I would suggest a separate PR on this once we are done with the wording update

> In order to determine if they have the necessary legal basis to process a user’s personal data for a specific purpose registered in the GVL, vendors must apply all relevant signals from the TC String in accordance to their registration. 

Reviewer: ```suggestion
In order to determine if they have the necessary legal basis to process a user’s personal data for a specific purpose registered in the GVL, vendors must follow all relevant signals from the TC String in accordance with what they disclose in the GVL. 
```

Author: Works, using registered twice is not good anyhow

> The relevant signals are: The applicable version of the GVL, the publisher restriction signal, the purpose legal basis signal and the vendor legal basis signal.

Reviewer: ```suggestion
The relevant signals in the TC string are the GVL version, the publisher restrictions signal, the purpose legal basis signal and the vendor legal basis signal.
```

Author: Work, makes the relation to the TC String more explicit

> Prior to processing a user’s personal data for a purpose registered in the GVL, vendors must:

Reviewer: ```suggestion
Prior to processing a user’s personal data for a purpose registered in the GVL, for each purpose vendors must:
```

Author: Good clarification

**Evaluate publisher restrictions** prior to any other signal:

> 1.  Check if the publisher disallowed the processing for this purpose using a purpose restriction 

Reviewer: ```suggestion
1.  Check if the publisher completely disallowed processing based on this purpose using a purpose restriction 
```

Author: Works for me, for/based on is something for native speaker ;)

> 2.  Check if the publisher specified the applicable legal basis for this purpose using a legal basis restriction:

Reviewer: ```suggestion
2.  Check if the publisher restricted the applicable legal basis for this purpose using a legal basis restriction (e.g. the publisher only allows "consent" as legal basis for this purpose, or only allows "legitimate interest" as legal basis for this purpose):
```

Author: Okay - I don't really like the restriction language, but given its how its named overall lets use it here too

>     -  In the absence of a legal basis restriction vendors must apply their default legal basis

Reviewer: ```suggestion
    -  In the absence of a legal basis restriction, vendors must apply their default legal basis
```

>     -  If there is a legal basis restriction vendors must apply the publisher defined legal basis 

Reviewer: ```suggestion
    -  If there is a legal basis restriction vendors must apply the publisher defined legal basis, which is only allowed if that purpose is either declared with the allowed legal basis as default legal basis, or if that purpose was declared as flexible
```

Author: Fine - can't be said often enough ;)

> In case the publisher disallows processing or the publisher defined legal basis is not registered by the vendor nor is the vendor flexible it may not process for that purpose. For example, if a vendor registered legitimate interest as legal basis for a purpose and is not flexible it may not process if there is a legal basis restriction to require consent.

Reviewer: ```suggestion
In case the publisher disallows processing or in case the acceptable legal basis defined by the publisher in the restrictions is not registered by the vendor as default purpose nor is the vendor declaring that purpose as flexible, the vendor may not process personal data based on that purpose. For example, if a vendor registered legitimate interest as legal basis for a purpose and is not declaring legal basis for that purpose as flexible, it may not process in the presence of a legal basis restriction that requires consent.
```

Author: Fine with me, more elaborate description

> In case the publisher disallowed processing or the publisher defined legal basis is not registered by the vendor nor is the vendor flexible it may not process for that purpose. As an example, if a vendor registered legitimated interest as legal basis for a purpose and is not flexible it may not process if there is a legal basis restriction to "require consent".

Reviewer 2: First sentence:
change "disallowed" to "disallows" (prefer present tense where possible)
Second sentence:
change "As an example" to "For example"
change "legitimated" to "legitimate" 
remove "" from "require consent"

**Evaluate purpose and vendor legal basis** 

> After determining the applicable legal basis vendors must then check

Reviewer: ```suggestion
After determining the applicable legal basis, vendors must then check:
```

> 1. the presence of a purpose legal basis signal for the respective purpose 

Reviewer: ```suggestion
1. the presence of a purpose legal basis signal for each purpose 
```

2. the presence of a vendor legal basis signal 

> Only if both signals are present for the applicable legal basis in the TC String may the vendor process for that purpose.

Reviewer: ```suggestion
Only if both signals are positive for the applicable legal basis in the TC String may the vendor process for that purpose.
```

> Only if both signals are present for the applicable legal basis in the TC String, the vendor may then process for that purpose.

Reviewer 2: Reword sentence to:
Only if both signals are present for the applicable legal basis in the TC String  may the vendor process for that purpose.

Author: Thx. works

## How to determine if data may be transmitted?<a name="handletcstring"></a>

> According to the policies of the Transparency and Consent Framework, a vendor may choose not to transmit data to another vendor for any reason, but a vendor must not transmit data to another vendor without a justified basis for relying on that vendor’s legal basis for processing the personal data. If a vendor has or obtains personal data and has no legal basis for the access to and processing of that data, the vendor should quickly cease collection and storage of the data and refrain from passing the data on to other parties, even if those parties have a legal basis. To determine if a vendor has at least one legal basis to process a user’s personal data [see](#detlegalbasis).

Reviewer 2: I just noticed two things:
# the last sentence is incomplete
# "see" link is broken

Reviewer 2: @asr-enid - what did you mean to link to?

Author: Checking

Author: I just wanted to link to the previous paragraph, given the Vendor A would check for Vendor B as described in how to determine a legal basis.

Reviewer 2: Ok, I can make this change.

Author: Okay thanks

> According to policies of the Transparency and Consent Framework, a vendor may choose not to transmit data to another vendor for any reason, but a vendor must not transmit data to another vendor without a justified basis for relying on that vendor’s legal basis for processing the personal data. If a vendor has or obtains personal data and has no legal basis for the access to and processing of that data, the vendor should quickly cease collection and storage of the data and refrain from passing the data on to other parties, even if those parties have a legal basis. To determine if a vendor has at least one legal basis to process a user’s personal data [see](#detlegalbasis).

Reviewer 2: First sentence:
add "the" in front of "policies"
In general:
This block of text should be reviewed by IAB Europe. I believe it is accurate, but it's good to be sure that the language matches what can be found in the policies.

Reviewer 2: I'm actually seeing that this is how the text currently reads. I do not love it, but you can disregard my comment ablve.

Author: Hi Alex - I only adapted they heading of this, this part is already in the current implementation guideline.

> According to the policies of the Transparency and Consent Framework, a vendor may choose not to transmit data to another vendor for any reason, but a vendor must not transmit data to another vendor without a justified basis for relying on that vendor’s legal basis for processing the personal data. If a vendor has or obtains personal data and has no legal basis for the access to and processing of that data, the vendor should quickly cease collection and storage of the data and refrain from passing the data on to other parties, even if those parties have a legal basis. To determine if a vendor has at least one legal basis to process a user’s personal data [see "How to determine legal bases from the TC String?"](#detlegalbasis).

Reviewer: Not very informed about this bit, could you share the policy references? Otherwise I'll leave it to others to check.

Author: @Facens this is an old text element that was referring to passing a signal, this commit only changes the title.
